api/src/main/java/org/openmrs/api/UserService.java

```diff
@@ -110,12 +111,12 @@
 	public User getUserByEmail(String email);
 	
 	/**
-	 * Gets user using user activation key
-	 * @param activationKey User's activation(token) for password reset 
-	 * @return requested user
+	 * Gets Login Credential using user activation key
+	 * @param token User's (token) for password reset 
+	 * @return requested LoginCredential with associated hashed of token as activation key
 	 */
 	@Authorized( { PrivilegeConstants.GET_USERS })
-	public User getUserByActivationKey(String activationKey);
```

Comment:
You should keep getUserByActivationKey, LoginCredential is not part of the public facing API so you should not return it or expose it

Reply:
To be clear get User by activation key should stay, get login credential by key should be in the DAO, get User by key delegating to the DAO

Reply:
The query for getting user by activation key started failing because the mapping was removed and as well the  activation key from the user object. Could you please provide a way of querying using the activation key>

Reply:
Since you moved the field to LoginCredential then you have to change the query

Reply:
I did change the query  just couldn't get one that uses creteria and get the user by activationkey.

Reply:
what is the query you are using?

Reply:
And recall what saying mostly here is you should not expose getLoginCredentialBykey and not how you should write the query

Reply:
i tried this (User) sessionFactory.getCurrentSession().createQuery("from User u where u.activation_key = :activationKey").setString(
			    "activationKey", activationKey).uniqueResult(); it didn't also turn out to work

Reply:
I just mentioned above that since you moved the field to LoginCredential class that that should what query needs to search from

Reply:
Ok When you said i should allow getUserByAcivationKey. My issue is i am unable to get user by activation key directly as you specified. Should  i just allow the getUserByActivation key and internally in the DAO layer i use getLoginCredentialByToken which i already implimented then using the id of the user from the LoginCredentialreturned i can now retreive the real user Instance which can be exposed.

Reply:
I don't seem to understand what you are asking. All am saying is that you need to have getUserByActivationKey only on the service and then you should have getLoginCredentialByKey in the DAO, your service method delegate to the DAO method. LoginCredential has a userId field that you should use to load the use within the service.